easybuild/main.py

```diff
@@ -390,6 +389,11 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None):
     if try_to_generate and build_specs and not generated_ecs:
         easyconfigs = tweak(easyconfigs, build_specs, modtool, targetdirs=tweaked_ecs_paths)
 
+    if options.containerize:
```

Comment:
I think you should put this after resolving dependencies so you will be able to extract the os dependencies for all the applications that should be build and not only the user requested one.

Reply:
Ah, that's indeed a good point...

But then we'll need to discriminate between the easyconfigs that were specified initially, and the full set of easyconfigs after resolving dependencies.

OK if we leave this for later enhancement? Maybe open an issue on this at https://github.com/easybuilders/easybuild-framework/issues so we won't forget?

Reply:
I was thinking about just passing the passed arguments/options to the `eb` inside the recipe but only removing the container specific options.

Reply:
That makes sense, that's also what we do with `--job` (except that there we also remove `--robot` since each job takes care of one installation and dependencies are set between them), see https://github.com/easybuilders/easybuild-framework/blob/master/easybuild/tools/parallelbuild.py#L129

Reply:
Follow-up in https://github.com/easybuilders/easybuild-framework/issues/2475